tests/diff/patch.c

```diff
@@ -610,3 +610,46 @@ void test_diff_patch__line_counts_with_eofnl(void)
 
 	git_buf_free(&content);
 }
+
+void test_diff_patch__can_strip_bad_utf8(void)
+{
+	const char *one_sha = "108e5f09fb1af8b969df1ed92e446a1e39d2c036";
+	const char *another_sha = "cdd42da2d1d53d1db2c98b7d4e972b450601c1ef";
+	git_tree *one, *another;
+	git_diff *diff;
+	git_patch *patch;
+	git_buf buf = GIT_BUF_INIT;
+	const char *expected = "diff --git a/Japanese.md b/Japanese.md\n"
+		"index dd469c9..caf022d 100644\n"
+		"--- a/Japanese.md\n"
+		"+++ b/Japanese.md\n"
+		"@@ -24,7 +24,7 @@ Redactedでは**レイヤ**における**zIndex**属性を制御可能にする \n"
```

Comment:
I'm a bit surprised there's a space right in front of the newline. As the test should actually show that we're correctly truncating UTF8 at the context line's cutoff point, I'd have expected the last character before the NL to be a UTF8 code point.

Reply:
I am confused too. I verified that the buffer truncates the length properly; not sure where this extra space comes from yet. I'll have to dig into it later.

Reply:
Ok, I figured it out. Sanitizing the buffer for valid UTF-8 characters will likely remove the newline from the hunk header, so we need to add it back if it was stripped.